### PR TITLE
Add test case for "missing scalafmt version"

### PR DESCRIPTION
**Pull request by olafurpg**

### CODE REVIEW

File: tests/unit/src/test/scala/tests/FormattingSlowSuite.scala [hunk 1]
```diff
@@ -225,4 +226,40 @@ object FormattingSlowSuite extends BaseSlowSuite("formatting") {
...
+        MissingScalafmtVersion.fixedVersion.getMessage
+      )
+      _ = assertNoDiff(client.workspaceDiagnostics, "")
+      // check file was not formatted because version was missing.
+      _ = assertNoDiff(
+        server.bufferContent("Main.scala"),
+        "object   Main"
+      )
+      _ <- server.formatting("Main.scala")
+      // check file was formatted now that version has been added.
```
Comment (gabro) [inline tests/unit/src/test/scala/tests/FormattingSlowSuite.scala]: Two things:

- I don't understand (reading the test), where is the interaction that adds the version
- I would also test against the content of `.scalafmt.conf`, which should now contain the version

Comment (olafurpg) [reply tests/unit/src/test/scala/tests/FormattingSlowSuite.scala]: Good points, will update.

**Comment by olafurpg** [comment]
I understand the error now, it's legitimate. The `formatting` request future returns before we fix the file, they in parallel asynchronously. Trying to figure out how to make it robust without a Thread.sleep

**Comment by olafurpg** [comment]
The last commit is pretty ugly, but it adds a nice improvement where we format the source file after creating the file or updating the version avoiding the need to notify the user they should "try formatting again".

File: tests/unit/src/test/scala/tests/FormattingSlowSuite.scala [hunk 2]
```diff
@@ -225,4 +229,99 @@ object FormattingSlowSuite extends BaseSlowSuite("formatting") {
...
+      _ <- server.initialize(
+        """|/.scalafmt.conf
+           |maxColumn=40
+           |/Main.scala
+           |object   Main
+           |""".stripMargin,
+        expectError = true
+      )
+      _ <- server.didOpen("Main.scala")
+      _ <- server.formatting("Main.scala")
```
Comment (olafurpg) [inline tests/unit/src/test/scala/tests/FormattingSlowSuite.scala]: To explain the problem, this future here returned before we were able to fix `.scalafmt.conf`. Now the `formatting` future includes waiting for the "update .scalafmt.conf?" dialogue.

**Comment by olafurpg** [comment]
Hurray, CI is stable now 🎉

**Pull request merged**